orchestrator/fork.go

```diff
@@ -0,0 +1,105 @@
+package orchestrator
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	registry "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum-optimism/supersim/config"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+const blockTime = 2
+
+func ConfigFromForkCLIConfig(cfg *config.ForkCLIConfig) (*OrchestratorConfig, error) {
+	superchain := registry.Superchains[cfg.Network]
+	chains := []ChainConfig{}
+
+	// L1
+	l1Client, err := ethclient.Dial(superchain.Config.L1.PublicRPC)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial l1 public rpc: %w", err)
+	}
+
+	l1ForkHeight := big.NewInt(rpc.FinalizedBlockNumber.Int64())
+	if cfg.L1ForkHeight > 0 {
+		l1ForkHeight = new(big.Int).SetUint64(cfg.L1ForkHeight)
+	}
```

Comment:
should this come before `rpc.FinalizedBlockNumber`? that way if `cfg.L1ForkHeight` is set we avoid having to make the rpc for the finalized block number?

Reply:
so we don't actually end up making an extra rpc here!

this just sets up the label to be used for the rpc on line 32. If `cfg.L1ForkHeight` is set, it'll override `l1ForkHeight` from the finalized label to the block number